applications/HDF5Application/custom_io/hdf5_container_component_io.cpp

```diff
@@ -389,477 +96,331 @@ ContainerComponentIO<TContainerType, TContainerItemType, TComponents...>::Contai
-            rData(i, j) = r_data[j];
-    }
+        KRATOS_ERROR_IF(rAttributesMap.find(rComponentName) != rAttributesMap.end())
+            << "The attributes for component \"" << rComponentName << "\" already exists.";
 
-    KRATOS_CATCH("");
-}
+        auto attributes = mpFile->ReadAttribute(r_data_set_path);
 
-template <typename TContainerItemType, typename TDataType>
-void SetItemDataValues(Variable<TDataType> const& rVariable,
-                       Vector<TDataType> const& rData,
-                       std::vector<TContainerItemType*>& rContainerItems)
-{
-    KRATOS_TRY;
+        if constexpr(value_type_traits::IsDynamic) {
+            std::vector<unsigned int> shape;
+            Internals::GetShapeFromAttributes(shape, attributes);
 
-    KRATOS_ERROR_IF(rContainerItems.size() != rData.size())
-        << "Number of items does not equal data set dimension\n"
-        << "[ rVariable = " << rVariable.Name() << ", rContainer.size() = "
-        << rContainerItems.size() << ", ReadData.size() = "
-        << rData.size() << " ].\n";
-    for (std::size_t i = 0; i < rContainerItems.size(); ++i)
-        rContainerItems[i]->GetValue(rVariable) = rData[i];
+            Matrix<value_primitive_type> values;
+            values.resize(BlockSize, std::accumulate(shape.begin(), shape.end(), 1U, std::multiplies<unsigned int>{}));
+            mpFile->ReadDataSet(r_data_set_path, values, StartIndex, BlockSize);
+            Internals::CopyFromContiguousDataArray(rLocalContainer, r_component, rContainerDataIO, DataTypeTraits<Matrix<value_primitive_type>>::GetContiguousData(values), shape);
+        } else {
+            Vector<value_type> values;
+            values.resize(BlockSize);
+            mpFile->ReadDataSet(r_data_set_path, values, StartIndex, BlockSize);
+            Internals::CopyFromContiguousDataArray(rLocalContainer, r_component, rContainerDataIO, DataTypeTraits<Vector<value_type>>::GetContiguousData(values), value_type_traits::Shape(value_type{}));
+        }
 
-    KRATOS_CATCH("");
-}
+        NewContainerComponentIOUtilities::SynchronizeComponent<TContainerType>::template Execute<TContainerDataIO>(rCommunicator, r_component);
 
-template <typename TContainerItemType>
-void SetItemDataValues(Flags const& rVariable,
-                       Vector<int> const& rData,
-                       std::vector<TContainerItemType*>& rContainerItems)
-{
-    KRATOS_TRY;
+        if (attributes.Has("__data_dimension")) attributes.RemoveValue("__data_dimension");
+        if (attributes.Has("__data_shape")) attributes.RemoveValue("__data_shape");
+        if (attributes.Has("__container_type")) attributes.RemoveValue("__container_type");
+        if (attributes.Has("__data_name")) attributes.RemoveValue("__data_name");
+        if (attributes.Has("__data_location")) attributes.RemoveValue("__data_location");
+        if (attributes.Has("__mesh_location")) attributes.RemoveValue("__mesh_location");
```

Comment:
these should be in some central location so if we ever need to update them, we can do so more easily (and they'd also be a bit easier to look through), Also, `ContainerComponentIO`'s docs should mention these reserved attributes.

Reply:
done :)